src/libraries/Microsoft.Extensions.FileSystemGlobbing/tests/FilePatternMatchTests.cs

```diff
@@ -28,13 +28,10 @@ public void TestGetHashCode()
         [Fact]
         public void TestGetHashCodeWithNull()
         {
-            FilePatternMatch match = new FilePatternMatch(null, null);
-            Assert.Equal(0, match.GetHashCode());
+            const string nonNullString = "non null";
+            int hash1 = new FilePatternMatch(nonNullString, null).GetHashCode();
```

Comment:
I'm not sure what guarantees these hashes to be non-equal (same comment applies before this change). It looks to me this test will very rarely randomly fail.

Reply:
I guess it "Different value == different hash" logic. How can it fail if both data and hashing algorithm are constant? If algorithm will change, this might fail, but it would always fail in that case.

Reply:
nope - different value => most likely different hash, hash is only 32-bit but string can have way more values than that. You should not rely on that behavior - this test will most likely be flaky and will be failing super rarely but still

Reply:
The only guarantees you have is that hash will be same for same value. I'd rather have this test removed than failing once every 2 years